# Unicorn Companies - Data Visualisation 
__________________________________________

### Data Visualisation 3 - Final Project
__________________________________________

## Data and Analytical Questions

The dataset is obtained form *__[Mavren Analytics](https://www.mavenanalytics.io/data-playground)__*, there are a bunch of really good datasets that you can work with and upscale your data analytics skills. I downloaded the dataset and saved into my personal GitHub repository. You can directly access the data and colab notebook by clicking *__[HERE](https://github.com/alisial94/DV3_DataVisualisation_UnicornCompanies)__*. 

Unicorn Dataset, the one I am working with, comprises of 1074 private companies with a valuation over $1 billion as of March 2022, including each company's current valuation, funding, country of origin, industry, select investors, and the years they were founded and became unicorns. It would be interseting to see how much have these countries grown in valuation compared to funding the recieved to start. Below you can find a list of analytical questions I would like to answer as part of the analysis and these questions are provided by Mavern Analytics. 

- Which unicorn companies have had the biggest return on investment?
- How long does it usually take for a company to become a unicorn? Has it always been this way?
- Which countries have the most unicorns? Are there any cities that appear to be industry hubs?
- Which investors have funded the most unicorns?

now let's jump into analysis and lets try answer these questions. 


 ## Data Cleaning & Exploratory Analysis

In this section we will be exploring the data to see how it looks and if it requires any transformations before we proceed to analysising the questions. After obtaining the final dataset, I'll also perform some basic EDA. 

### Importing required libraries

In [ ]:
# importing libraries
import pandas as pd
import plotly.express as px
import numpy as np

# install dash app
! pip install jupyter-dash # For Dash applications


# import the Dash core and HTML components
from dash import dcc
from dash import html

# we need to import the input and output functions for the interactive features
from dash import Input, Output


# import the JupyterDash extension
from jupyter_dash import JupyterDash

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


### Loding the Data

In [ ]:
# loading the data from the git repo

df = pd.read_csv('https://raw.githubusercontent.com/alisial94/DV3_DataVisualisation_UnicornCompanies/main/Data/Unicorn_Companies.csv')
df.head()

Company Valuation Date Joined                         Industry  \
0  Bytedance     $180B  2017-04-07          Artificial intelligence   
1     SpaceX     $100B  2012-12-01                            Other   
2      SHEIN     $100B  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe      $95B  2014-01-23                          Fintech   
4     Klarna      $46B  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded Funding  \
0        Beijing          China           Asia          2012     $8B   
1      Hawthorne  United States  North America          2002     $7B   
2       Shenzhen          China           Asia          2008     $2B   
3  San Francisco  United States  North America          2010     $2B   
4      Stockholm         Sweden         Europe          2005     $4B   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

In [ ]:
# checking the data type of each column to fix it according to our requirement

df.dtypes

Company             object
Valuation           object
Date Joined         object
Industry            object
City                object
Country             object
Continent           object
Year Founded         int64
Funding             object
Select Investors    object
dtype: object

### Transforming the Data 

It is clear that most of our columns are strings including the valuationa and funding columns. To preceed any further we needs the columns in integer format. There are few companies whose funding is unknow, I'll be removing them from the dataset for obvious reasons. Next i will be adjusting the date column to datatime format and changing the year column to integer. I'll also be spliting the investor column into three new columns

#### 1 - Valuation Column

In [ ]:
# fixing the column type of valuation

# replacing strings
df['Valuation'] = df['Valuation'].str.replace("$","")
df['Valuation'] = df['Valuation'].str.replace("B","000000000")

# changing type to int
df['Valuation'] = df['Valuation'].astype(int)

df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded Funding  \
0        Beijing          China           Asia          2012     $8B   
1      Hawthorne  United States  North America          2002     $7B   
2       Shenzhen          China           Asia          2008     $2B   
3  San Francisco  United States  North America          2010     $2B   
4      Stockholm         Sweden         Europe          2005     $4B   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

#### 2 - Funding 

In [ ]:
# fixing the column type of Funding

# replacing strings
df = df[df['Funding'] != 'Unknown']
df = df[df['Funding'] != '0']
df['Funding'] = df['Funding'].str.replace("$","")
df['Funding'] = df['Funding'].str.replace("B","000000000")
df['Funding'] = df['Funding'].str.replace("M","000000")


# changing type to int
df['Funding'] = df['Funding'].astype(int)

df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  
0  Sequoia Capital China, SIG Asia Investments, S...  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...  
2  Tiger Global Management, Sequoia Capital China...  
3        Khosla Ventures, LowercaseCapital, capitalG  
4  Institutional Venture Partners, Sequoia Capita...

In [ ]:
# one company has a value of zero, and was formed in 1919, i'll be removing this as well.
df[df['Company'] == 'Otto Bock HealthCare']

Company   Valuation Date Joined Industry        City  \
189  Otto Bock HealthCare  4000000000  2017-06-24   Health  Duderstadt   

     Country Continent  Year Founded  Funding Select Investors  
189  Germany    Europe          1919        0     EQT Partners

In [ ]:
df = df[df['Company'] != 'Otto Bock HealthCare']

#### 3 - Date Joined

I'll also be extracting the year from date joined column to you later in analysis.

In [ ]:
# transforming Date Joined column

df['Date Joined'] = pd.to_datetime(df['Date Joined'], format='%Y-%m-%d')
df['Year Joined'] = pd.DatetimeIndex(df['Date Joined']).year



df.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  
0  Sequoia Capital China, SIG Asia Investments, S...         2017  
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012  
2  Tiger Global Management, Sequoia Capital China...         2018  
3        Khosla Ventures, LowercaseCapital, capitalG         2014  
4  Institutional Venture Partners, Sequoia Capita...         2011

#### 4 - Investors

Why I plan to split the investor column into 4, when I looked and explored the investor column and I realised that at max there are 4 investors involved in the funding of one company. Thus, 4 columns were made to later explore the count of investors. 

In [ ]:
# splitiing investors into 4 columns

df[['First Investor','Second Investor', 'Third Investor','Fourth Investor']] = df['Select Investors'].str.split(',', n=3, expand=True)


# removing any leading and trailing space
df['First Investor'] = df['First Investor'].str.strip()
df['Second Investor'] = df['Second Investor'].str.strip()
df['Third Investor'] = df['Third Investor'].str.strip()
df['Fourth Investor'] = df['Fourth Investor'].str.strip()
df.head()



Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor  
0                Sina Weibo  Softbank Group  
1       Rothenberg Ventures            None  
2  Shunwei Capital Partners            None  
3                  capitalG            None  
4          General Atlantic            None

### Final Dataset

As you just saw above, our final dataset contains a total of 15 columns apart from index and at this point I wont be dropping any columns. Now lets check out some visualistaions 

In [ ]:
df.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor  
0                Sina Weibo  Softbank Group  
1       Rothenberg Ventures            None  
2  Shunwei Capital Partners            None  
3                  capitalG            None  
4          General Atlantic            None

### Top 10 companies based on valuation

In [ ]:
# Looking at the arranging the dataset

df_top10_val = df.sort_values('Valuation', axis = 0, ascending = False).head(10)

# creating the lable 
labels_1 = {'Company': 'Company Name', 'Valuation':"Total Valuation"}

# ploting the graph
Top_10_hist = px.histogram(df_top10_val, x='Company', y='Valuation',
                      color = 'Company', 
                      template='simple_white', 
                      title = "Top 10 Companies by Valuation",
                      labels = labels_1,
                      hover_name= 'Company', 
                      hover_data={'Company' : False,
                                  'Year Founded' : True,
                                  'Funding' : True,
                                  'Valuation' : False})

Top_10_hist.update_layout( yaxis_title="Total Valuation", hovermode="x unified")
                          

Top_10_hist.show()

## Question 1 - Which unicorn companies have had the biggest return on investment?

For this question we will be calculating a new field in our dataset which with be call ROI (Retuen on Investment). Thes field will be calculated using Funding amount and valuation amount.

### Adding the new calculated field 

In [ ]:
# calculating ROI
df['ROI'] = (df['Valuation'] - df['Funding'])/df['Funding']

# changing it to two decimal places
df = df.round(2)
df.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  
0                Sina Weibo  Softbank Group  21.50  
1       Rothenberg Ventures            None  13.29  
2  Shunwei Capital Partners            None  49.00  
3                  capitalG            None  46.50  
4          General Atlantic            None  10.50

### Ploting the Top 10 Companies in terms of ROI

In [ ]:
# Looking at the arranging the dataset

df_top10_roi = df.sort_values('ROI', axis = 0, ascending = False).head(10)

# creating the lable 
labels_2 = {'Company':'Company Name', 'ROI':"Return on Investment"}

# ploting the graph
Top_10_roi_hist = px.histogram(df_top10_roi, x='Company', y='ROI',
                      color = 'Company', 
                      template='simple_white', 
                      title = "Top 10 Companies by Return on Investment",
                      labels = labels_2,
                      hover_name= 'Company', 
                      hover_data={'Company' : False})

Top_10_roi_hist.update_layout( yaxis_title="Total Return on Investment", hovermode="x unified")
                          

Top_10_roi_hist.show()

Based on the graph above we can definately say that Zapier appears to have the highest return on investment from the list of companies remaining in our dataset. Including Zapier has resulted in a poor performing graph as we can visualise the difference betwee the ROIs of other nine companies. In the next graph you will see Top 10 companies with Zapier. 

In [ ]:
# Looking at the arranging the dataset

df_top10_roi_noZap = df.sort_values('ROI', axis = 0, ascending = False).head(11)
df_top10_roi_noZap = df_top10_roi_noZap[df_top10_roi_noZap['Company'] != 'Zapier']

# creating the lable 
labels_2 = {'Company':'Company Name', 'ROI':"Return on Investment"}

# ploting the graph
Top_10_roi_hist = px.histogram(df_top10_roi_noZap, x='Company', y='ROI',
                      color = 'Company', 
                      template='simple_white', 
                      title = "Top 10 Companies by Return on Investment (without Zapier)",
                      labels = labels_2,
                      hover_name= 'Company', 
                      hover_data={'Company' : False})

Top_10_roi_hist.update_layout( yaxis_title="Total Return on Investment", hovermode="x unified")
                          

Top_10_roi_hist.show()

## Question 2 - How long does it usually take for a company to become a unicorn? Has it always been this way?

This is a tricky question I would say, because your approach for final result will have a very subjective route. Why is that simply becuase you will have to make some arbitary assumstions to calculate the time it took for a company to grow and reach $1 Billion in valuation. 

The first thing I explored was the distinct funding values below $500 Million to capture the impact of growth. This value could have been something else based on what you assume would be a good valuation to begin with. Next I looked up the difference between the year each of these company were founded and the year the reach 1 Billion in valuation. Below you will see the code I used to achieve this.   

### Caluation Difference in year

In [ ]:
# filter the data to $500 Million starting capital

df_fund500 = df[df['Funding'] != 500000000]
df_fund500.head()

Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  
0                Sina Weibo  Softbank Group  21.50  
1       Rothenberg Ventures            None  13.29  
2  Shunwei Capital Partners            None  49.00  
3                  capitalG            None  46.50  
4          General Atlantic            None  10.50

In [ ]:
# adding the calculated field for year to grow to 1 Billion

df_fund500['Growth Year'] = df_fund500['Year Joined'] - df_fund500['Year Founded']
df_fund500.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Company     Valuation Date Joined                         Industry  \
0  Bytedance  180000000000  2017-04-07          Artificial intelligence   
1     SpaceX  100000000000  2012-12-01                            Other   
2      SHEIN  100000000000  2018-07-03  E-commerce & direct-to-consumer   
3     Stripe   95000000000  2014-01-23                          Fintech   
4     Klarna   46000000000  2011-12-12                          Fintech   

            City        Country      Continent  Year Founded     Funding  \
0        Beijing          China           Asia          2012  8000000000   
1      Hawthorne  United States  North America          2002  7000000000   
2       Shenzhen          China           Asia          2008  2000000000   
3  San Francisco  United States  North America          2010  2000000000   
4      Stockholm         Sweden         Europe          2005  4000000000   

                                    Select Investors  Year Joined  \
0  Sequoia Capital China, SIG Asia Investments, S...         2017   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...         2012   
2  Tiger Global Management, Sequoia Capital China...         2018   
3        Khosla Ventures, LowercaseCapital, capitalG         2014   
4  Institutional Venture Partners, Sequoia Capita...         2011   

                   First Investor          Second Investor  \
0           Sequoia Capital China     SIG Asia Investments   
1                   Founders Fund  Draper Fisher Jurvetson   
2         Tiger Global Management    Sequoia Capital China   
3                 Khosla Ventures         LowercaseCapital   
4  Institutional Venture Partners          Sequoia Capital   

             Third Investor Fourth Investor    ROI  Growth Year  
0                Sina Weibo  Softbank Group  21.50            5  
1       Rothenberg Ventures            None  13.29           10  
2  Shunwei Capital Partners            None  49.00           10  
3                  capitalG            None  46.50            4  
4          General Atlantic            None  10.50            6

Now that we have the number of years it took for these companies to enter the category of unicorn. Next i'll be taking an average of this new calculated feild to find out how much time on average would it take for any company to grwow aand become unicorn (reach 1 Billion). The calculation is performed below but the value comes up to be around 7 years. This value, as I said earlier, is arbirtary and resulted due assumptions I made.

In [ ]:
# Average number of years it takes a company to become Unicorn

df_fund500['Growth Year'].mean()


6.906515580736544

### Second part of the this question

Even though we have an answer for the average number of years it takes a company to become unicorn, the second part asks us to inlude time as categorical variable. What I mean is that use time split data and compair if the value we calculated earlier changes in a decade or whatever timeframe you wish to divide the data in. I'll be dividing the data as follow:

- 1990-1999
- 2000-2009
- 2010-2020

In this way i'll three decades and their averages to comapair. Just to go over this again, we'll divide the data in to decades based on the year of each these companies were founded in and then take the difference with the year the reached the unicorn threshold.

In [ ]:
# Dividing the data

# 1990-1999
df_90 = df_fund500[(df_fund500['Year Founded'] >=1990) & (df_fund500['Year Founded'] < 2000)] 

# 2000-2009
df_00 = df_fund500[(df_fund500['Year Founded'] >=2000) & (df_fund500['Year Founded'] < 2010)] 

# 2010-2019
df_10 = df_fund500[(df_fund500['Year Founded'] >=2010) & (df_fund500['Year Founded'] < 2020)] 


In [ ]:
# average years for companies formed in 90s

avg_90 = df_90['Growth Year'].mean()
avg_90

21.391304347826086

In [ ]:
# average years for companies formed in 2000s

avg_00 = df_00['Growth Year'].mean()
avg_00

12.771812080536913

In [ ]:
# average years for companies formed in 2010 onwards

avg_10 = df_10['Growth Year'].mean()
avg_10


5.669411764705882

In [ ]:
# creating a new dataframe for visualisations

# initialize list of lists
data = [['1990-99', avg_90], ['2000-10', avg_00], ['2010-20', avg_10]]

# Create the pandas DataFrame
decade_avg = pd.DataFrame(data, columns=['Decade', 'Avg Years']).round(2)
decade_avg


Decade  Avg Years
0  1990-99      21.39
1  2000-10      12.77
2  2010-20       5.67

In [ ]:
# ploting the avg year graph 

# setting colours
#dec_col = ("1990-99" = '#3D9970', '2000-10'= '#FF4136', '2010-20'= '#FF851B')

decade_avg_hist = px.histogram(decade_avg, x='Decade', y='Avg Years',
                               color="Decade", 
                               color_discrete_map={"1990-99" : '#3D9970', "2000-10": '#FF4136', "2010-20": '#FF851B'}, 
                               template='simple_white', 
                               title = "Avg years by Deacades",
                               hover_name= 'Decade', 
                               hover_data={'Decade' : False})

decade_avg_hist.update_layout( yaxis_title="Average Years", hovermode="x unified")
                          

decade_avg_hist.show()




# Box plot for overall distribtion of number of years by the decades
import plotly.graph_objects as go

x1 = ['1990-99', '1990-99', '1990-99', '1990-99', '1990-99', '1990-99',
     '1990-99','1990-99','1990-99', '1990-99','1990-99', '1990-99','1990-99',
      '1990-99','1990-99','1990-99','1990-99','1990-99','1990-99','1990-99','1990-99',
      '1990-99',]



dec_avg = go.Figure()
dec_avg.add_trace(go.Box(x = x1, y=df_90['Growth Year'],
                     name='1990-99',
                     marker_color='#3D9970'))
dec_avg.add_trace(go.Box(y=df_00['Growth Year'],
                     name='2000-10',
                     marker_color='#FF4136'))
dec_avg.add_trace(go.Box(y=df_10['Growth Year'],
                     name='2010-20',
                     marker_color='#FF851B'))

dec_avg.update_layout( yaxis_title="Average Years", 
                  xaxis_title = "Decades", 
                  title = "Distribution of Number of Years to reach Unicorn threshold",
                  template='simple_white')

dec_avg.show()

Based on the graphs above we can see that the average number of years it took for a company to be recognised as Unicorn has reduced over time. This clearly indicates that now we live in a more faced paced global enviornment where companies have a higher opportunity to be successful and grow.  

## Question 3 - Which countries have the most unicorns? Are there any cities that appear to be industry hubs?



In this section we will look at number of Unicorn Companies in each country and try to also analyze if there are any cities that might appear to industry hubs. In this section i'll be creating a map to to highlight the countries with Unicorn Companies. In order to do that I have included another dataset that includes latitute and longitute data for the countries, I will be calling this directly from the project repository at GitHub.

### Number of Unicorn per country

In [ ]:
# taking a count of unicorn companies in each country and save them in a new dataset

df_country = df.groupby(['Country']).size().reset_index(name='Count of Unicorn').sort_values('Count of Unicorn',ascending = False)
df_country.head()

Country  Count of Unicorn
44   United States               555
9            China               170
19           India                65
43  United Kingdom                43
17         Germany                25

In [ ]:
# importing lat long dataset

lat_long_df = pd.read_csv('https://raw.githubusercontent.com/alisial94/DV3_DataVisualisation_UnicornCompanies/main/Data/countries_data.csv')
lat_long_df.head(10)

country_code   latitude  longitude               country
0           AD  42.546245   1.601554               Andorra
1           AE  23.424076  53.847818  United Arab Emirates
2           AF  33.939110  67.709953           Afghanistan
3           AG  17.060816 -61.796428   Antigua and Barbuda
4           AI  18.220554 -63.068615              Anguilla
5           AL  41.153332  20.168331               Albania
6           AM  40.069099  45.038189               Armenia
7           AN  12.226079 -69.060087  Netherlands Antilles
8           AO -11.202692  17.873887                Angola
9           AQ -75.250973  -0.071389            Antarctica

In [ ]:
# renaming the columns

lat_long_df.rename(columns = {'country_code':'Country Code', 'latitude':'Latitude', 'longitude':'Longitude', 'country':'Country'}, inplace = True)
lat_long_df.head(2)

Country Code   Latitude  Longitude               Country
0           AD  42.546245   1.601554               Andorra
1           AE  23.424076  53.847818  United Arab Emirates

In [ ]:
# merging the two columns

df_country = pd.merge(df_country,lat_long_df,on='Country')
df_country.head()

Country  Count of Unicorn Country Code   Latitude   Longitude
0   United States               555           US  37.090240  -95.712891
1           China               170           CN  35.861660  104.195397
2           India                65           IN  20.593684   78.962880
3  United Kingdom                43           GB  55.378051   -3.435973
4         Germany                25           DE  51.165691   10.451526

In [ ]:
# creating the map

map = px.scatter_geo(df_country , lat = 'Latitude' , lon = 'Longitude',
                     color="Count of Unicorn", # which column to use to set the color of markers
                     hover_name="Country", # column added to hover information
                     size="Count of Unicorn", # size of markers
                     projection="natural earth")

map.update_layout(
        title = 'Countries by Number of Unicorn companies<br>(Hover for county names)',
        geo = dict(
            scope = 'world',
            landcolor = 'rgb(217, 217, 217)'))

map.show()

Observing the simple map above it is very obvious that USA has the most number of Unicorn Countries, having a count 555 followed by China and India. It would be more interesting to see how the map looks like if we remove these top 3 countries from the equation. In the next section we will just be doing that to highlight other emerging markets in the world. 

In [ ]:
# creating the map without USA, China or India

# filter the data
df_country_notop3 = df_country[df_country['Country Code'] != 'US']
df_country_notop3 = df_country_notop3[df_country['Country Code'] != 'CN']
df_country_notop3 = df_country_notop3[df_country['Country Code'] != 'IN']

# creat map
map_1 = px.scatter_geo(df_country_notop3 , lat = 'Latitude' , lon = 'Longitude',
                     color="Count of Unicorn", # which column to use to set the color of markers
                     hover_name="Country", # column added to hover information
                     size="Count of Unicorn", # size of markers
                     projection="natural earth")

map_1.update_layout(
        title = 'Countries by Number of Unicorn companies with Top 3<br>(Hover for county names)',
        geo = dict(
            scope = 'world',
            landcolor = 'rgb(217, 217, 217)'))

map_1.show()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:5: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



### Cities vs Industry Hub

In this section, I plan to analyse if there are any cities that might become hubs for a particular sector of the industry. The idea here is to group by the industry count by city and then see if we have few winners. For this exercise I will be assuming that an industry hub is considered a city having 8 or more companies from the same industry

In [ ]:
# grouping the data by city and industy

df_city = df.groupby(['City', 'Industry']).size().reset_index(name='Count of Industry').sort_values('Count of Industry',ascending = False)
df_city = df_city.head(16)
df_city

City                         Industry  Count of Industry
403  San Francisco     Internet software & services                 54
400  San Francisco                          Fintech                 41
318       New York                          Fintech                 33
243         London                          Fintech                 24
321       New York     Internet software & services                 19
320       New York                           Health                 14
402  San Francisco                           Health                 12
29         Beijing  E-commerce & direct-to-consumer                 11
453       Shanghai            Auto & transportation                 10
48       Bengaluru     Internet software & services                  9
35         Beijing      Mobile & telecommunications                  9
314       New York                    Cybersecurity                  9
396  San Francisco                    Cybersecurity                  8
30         Beijing                           Edtech                  8
394  San Francisco          Artificial intelligence                  8
26         Beijing          Artificial intelligence                  8

In [ ]:
# tree map for city wide industy hubs

tree_map = px.treemap(df_city, path=["City", 'Industry'], values='Count of Industry',
                  color='Count of Industry',
                  color_continuous_scale='viridis',
                  hover_data= {'City':False,
                               'Industry':False}
                 )
tree_map.update_layout(margin = dict(t=50, l=25, r=25, b=25))
tree_map.show()

Based on the the treemap above we can say that we do see some cities as industry hubs, for example, San Francisco appears to be industry for Internet Software Services and also Fintech.   

## Question 4 - Which investors have funded the most unicorns?

If you rememver that during the EDA phase, I decided to split the investor column in four new columns, that was mainly done to answer this question now we will be uing those columns and find the investors with most number of investments.

In [ ]:
df_first_investor = df.groupby(['First Investor']).size().reset_index(name='Count of First').sort_values('Count of First',ascending = False)
df_first_investor.rename(columns = {'First Investor':'Investor'}, inplace = True)
#df_first_investor['Investor'] = df_first_investor['Investor'].astype(str)
df_first_investor.head(10)

Investor  Count of First
420        Sequoia Capital China              30
37           Andreessen Horowitz              29
11                         Accel              28
252             Insight Partners              22
334    New Enterprise Associates              16
419              Sequoia Capital              16
245               Index Ventures              14
484      Tiger Global Management              14
297  Lightspeed Venture Partners              13
441               SoftBank Group              13

In [ ]:
df_second_investor = df.groupby(['Second Investor']).size().reset_index(name='Count of Second').sort_values('Count of Second',ascending = False)
df_second_investor.rename(columns = {'Second Investor':'Investor'}, inplace = True)
df_second_investor.head()

Investor  Count of Second
531  Tiger Global Management               20
17                     Accel               18
466          Sequoia Capital               17
227         General Catalyst               16
289         Insight Partners               14

In [ ]:
df_third_investor = df.groupby(['Third Investor']).size().reset_index(name='Count of Third').sort_values('Count of Third',ascending = False)
df_third_investor.rename(columns = {'Third Investor':'Investor'}, inplace = True)
df_third_investor.head()

Investor  Count of Third
505      Tiger Global Management              19
432              Sequoia Capital              13
14                         Accel              13
35           Andreessen Horowitz              12
307  Lightspeed Venture Partners              11

In [ ]:
df_fourth_investor = df.groupby(['Fourth Investor']).size().reset_index(name='Count of Fourth').sort_values('Count of Fourth',ascending = False)
df_fourth_investor.rename(columns = {'Fourth Investor':'Investor'}, inplace = True)
df_fourth_investor.head(10)

Investor  Count of Fourth
0                           Accel                1
1                   Goldman Sachs                1
2               McKesson Ventures                1
3               Redpoint Ventures                1
4             Redpoint e.ventures                1
5  SDIC CMC Investment Management                1
6                  Softbank Group                1
7                   Spark Capital                1

In [ ]:
# merging the tables for final table

df_investors = pd.merge(df_first_investor,df_second_investor, how='left')
df_investors = df_investors.merge(df_third_investor, on='Investor', how='left')
df_investors = df_investors.merge(df_fourth_investor, on='Investor', how='left')
#df_investors = df_investors.set_index('Investor')
df_investors.head()

Investor  Count of First  Count of Second  Count of Third  \
0      Sequoia Capital China              30              7.0            10.0   
1        Andreessen Horowitz              29             12.0            12.0   
2                      Accel              28             18.0            13.0   
3           Insight Partners              22             14.0            10.0   
4  New Enterprise Associates              16              4.0             3.0   

   Count of Fourth  
0              NaN  
1              NaN  
2              1.0  
3              NaN  
4              NaN

In [ ]:
# removing NaN values and fixing the type of columns

df_investors['Count of Second'] = df_investors['Count of Second'].fillna(0).astype(int)
df_investors['Count of Third'] = df_investors['Count of Third'].fillna(0).astype(int)
df_investors['Count of Fourth'] = df_investors['Count of Fourth'].fillna(0).astype(int)
df_investors.head()

Investor  Count of First  Count of Second  Count of Third  \
0      Sequoia Capital China              30                7              10   
1        Andreessen Horowitz              29               12              12   
2                      Accel              28               18              13   
3           Insight Partners              22               14              10   
4  New Enterprise Associates              16                4               3   

   Count of Fourth  
0                0  
1                0  
2                1  
3                0  
4                0

In [ ]:
# calculating the top number of investments

df_investors['Total Investments'] = df_investors['Count of First'] + df_investors['Count of Second'] + df_investors['Count of Third'] + df_investors['Count of Fourth']
df_investors = df_investors.sort_values('Total Investments',ascending = False)
df_investors.head(15)

Investor  Count of First  Count of Second  \
2                         Accel              28               18   
7       Tiger Global Management              14               20   
1           Andreessen Horowitz              29               12   
0         Sequoia Capital China              30                7   
3              Insight Partners              22               14   
5               Sequoia Capital              16               17   
14             General Catalyst              10               16   
8   Lightspeed Venture Partners              13               10   
9                SoftBank Group              13               12   
6                Index Ventures              14                9   
10             Tencent Holdings              12                8   
15             General Atlantic               9                8   
28              Google Ventures               6               11   
12        Sequoia Capital India              11                8   
27                  IDG Capital               6                8   

    Count of Third  Count of Fourth  Total Investments  
2               13                1                 60  
7               19                0                 53  
1               12                0                 53  
0               10                0                 47  
3               10                0                 46  
5               13                0                 46  
14               8                0                 34  
8               11                0                 34  
9                9                0                 34  
6                9                0                 32  
10               9                0                 29  
15              11                0                 28  
28               8                0                 25  
12               6                0                 25  
27               9                0                 23

In [ ]:
# plotting the top 15 investor results

bubble_chart = px.scatter(df_investors.head(15), x='Investor', y="Total Investments",
	         size="Total Investments", color="Total Investments",
                 hover_name="Investor", size_max=60,
                 hover_data={'Investor' : False})
bubble_chart.update_layout( title = "Top 15 investors based on Investments",
                  template='simple_white')
bubble_chart.show()